query/branch/default_branch_blocks_force_push.sql

```diff
@@ -1,19 +1,19 @@
 select
   -- Required Columns
-  r.full_name as resource,
+  url as resource,
   case
-    when b.allow_force_pushes_enabled = 'false' then 'ok'
+    when (default_branch_ref -> 'branch_protection_rule') is null then 'info'
+    when (default_branch_ref -> 'branch_protection_rule' ->> 'allows_force_pushes') = 'false' then 'ok'
     else 'alarm'
   end as status,
-  r.full_name || ' default branch ' || r.default_branch ||
+  name_with_owner || ' default branch ' || (default_branch_ref ->> 'name') ||
   case
-    when b.allow_force_pushes_enabled = 'false' then ' prevents force push.'
-    when b.allow_force_pushes_enabled = 'true' then ' allows force push.'
+    when (default_branch_ref -> 'branch_protection_rule' ->> 'allows_force_pushes') = 'false' then ' prevents force push.'
+    when (default_branch_ref -> 'branch_protection_rule' ->> 'allows_force_pushes') = 'true' then ' allows force push.'
     -- If not false or true, then null, which means no branch protection rule exists
-    else ' is not protected.'
+    else ' is not protected, or you have insufficient permissions to see branch protection rules.'
```

Comment:
I can't remember exactly, but with the v3 API, was it easy to tell no branch protection vs. access denied on viewing branch protection rules? If so, does the v4 API make it harder to distinguish this difference?

Reply:
Yes, it returns null/empty whether there is no protection or the user doesn't have permission to view the protection.